File: examples/java/operations/steps/add-numbers.java
import software.amazon.lambda.durable.DurableContext;
import software.amazon.lambda.durable.DurableHandler;
import software.amazon.lambda.durable.StepContext;

public class AddNumbersExample extends DurableHandler<Object, Integer> {
    @Override
    public Integer handleRequest(Object input, DurableContext context) {
        int result = context.step("add_numbers", Integer.class,
            (StepContext ctx) -> 5 + 3);
        return result;
    }
}

File: examples/csharp/operations/steps/add-numbers.cs
using Amazon.Lambda.Core;
using Amazon.Lambda.DurableExecution;

public class AddNumbersExample
{
    public Task<DurableExecutionInvocationOutput> Handler(
        DurableExecutionInvocationInput input, ILambdaContext context)
        => DurableFunction.WrapAsync<object, int>(Workflow, input, context);

    private async Task<int> Workflow(object input, IDurableContext ctx)
    {
        int result = await ctx.StepAsync(
            async (_, _) => AddNumbers(5, 3),
            name: "add_numbers");
        return result;
    }

    private static int AddNumbers(int a, int b) => a + b;
}
